sdk/tests/core/main.c

```diff
@@ -21,7 +21,8 @@ int main()
   // negative numbers
   result += test_az_base64();
   result += test_az_context();
-#if !defined(__APPLE__) && defined(PLATFORM_POSIX) && defined(_az_MOCK_ENABLED)
+#if !defined(__APPLE__) && defined(PLATFORM_POSIX) && defined(_az_MOCK_ENABLED) \
+    && !defined(TRANSPORT_PAHO)
```

Comment:
It feels a bit too complicated to keep both TRANSPORT_PAHO and AZ_MQTT_TRANSPORT_IMPL. Can we remove TRANSPORT_PAHO?

Reply:
TRANSPORT_PAHO is used for the IoT samples unfortunately

Reply:
Yes, I understand that: I'm proposing we replace TRANSPORT_PAHO with AZ_MQTT_TRANSPORT_IMPL=PAHO_SYNC, even for IoT Hub samples.